sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -267,6 +267,9 @@ public HttpClient build() {
             }
         }
 
+        // Do not follow redirects by default. We should let the pipeline have more control of this.
+        httpClientBuilder.followRedirects(false);
```

Comment:
should we expose this as an option or configuration property? I case somebody took dependency on this somehow.

Reply:
If an external customer took dependency on it, we are going to break them never the less if we change the default settings. If they have to change their code we might as well recommend them to add redirect policy.

For our SDKs - we support both Netty and OkHttp AFAIK. Since Netty does not handle redirects by default an SDK is less likely to hit this.

Reply:
We usually offer some compat switch for behavior breaking changes like this so that customers can bring back old behavior with least effort. Config switch is usually easiest one as it doesn't require code change in case of emergency.

If we think that adding redirect policy is acceptable easy solution then we should at least document mitigation steps in changelog when calling out this change.

Reply:
That is a fair point. I am not sure which of the 2 options should we choose it likely depends on how popular OkHttpClient is for our customers.  /cc: @srnagar , @JonathanGiles

Reply:
Given that OkHttpClientBuilder allows you to [create a client from an instance of `OkHttpClient`](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java#L72) directly, users who want this behavior can configure their own OkHttpClient with automatic redirection enabled and create the builder. 

Given that OkHttp is not the default client and isn't widely used, compat switch may not be required.

Reply:
Updated the changelog.

Reply:
The behavior via [create a client from an instance of OkHttpClient](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java#L72) is not correct. We override the values set on the OkHttpClientBuilder in our builder which seems like an oversight and something we should fix. I will create a bug to do this separately.

In order to support the current scenario I am exposing a new property so someone who depends on the older behavior has a fallback in place.